File: python/올바른괄호.py
def solution(s):
    answer = True
    
    left = 0
    right = 0
    for i in range(len(s)):
        if s[i] == '(':
            left += 1
        else:
            right += 1
        if right > left:
            return False
    if left == right:
        return True
    return False

print(solution("()()"))
print(solution("(())()"))
print(solution(")()("))
print(solution("(()("))

File: JavaScript/올바른괄호.js
function solution(s){
    let leftCount = 0;
    let rightCount = 0;
    for (let i = 0; i < s.length; i++) {
        if (s[i] === '(') leftCount += 1;
        if (s[i] === ')') rightCount += 1;
        if (leftCount < rightCount) return false;
    }
    return leftCount === rightCount;
}

console.log(solution("(())()"));
console.log(solution(")()("));